renderdoc/driver/gl/gl_pixelhistory.cpp

```diff
@@ -234,33 +370,146 @@ bool PixelHistorySetupResources(WrappedOpenGL *driver, GLPixelHistoryResources &
   resources.primitiveIdFragmentShaderSPIRV = CreateSPIRVShader(eGL_FRAGMENT_SHADER, spirvSource);
   resources.primitiveIdFragmentShader = CreateShader(eGL_FRAGMENT_SHADER, glslSource);
-
+  resources.msCopyComputeProgram = CreateCShaderProgram(msCopySource);
+  resources.msCopyDepthComputeProgram = CreateCShaderProgram(msCopySourceDepth);
   return true;
 }
 
 bool PixelHistoryDestroyResources(WrappedOpenGL *driver, const GLPixelHistoryResources &resources)
 {
-  driver->glDeleteTextures(1, &resources.colorImage);
-  driver->glDeleteTextures(1, &resources.dsImage);
-  driver->glDeleteFramebuffers(1, &resources.frameBuffer);
   driver->glDeleteTextures(1, &resources.fullPrecisionColorImage);
   driver->glDeleteTextures(1, &resources.fullPrecisionDsImage);
   driver->glDeleteFramebuffers(1, &resources.fullPrecisionFrameBuffer);
   driver->glDeleteShader(resources.primitiveIdFragmentShader);
   driver->glDeleteShader(resources.primitiveIdFragmentShaderSPIRV);
-  driver->glDeleteTextures(1, &resources.depthImage);
-  driver->glDeleteTextures(1, &resources.stencilImage);
+  driver->glDeleteProgram(resources.msCopyComputeProgram);
+  driver->glDeleteProgram(resources.msCopyDepthComputeProgram);
+  driver->glDeleteBuffers(1, &resources.msCopyDstBuffer);
+  driver->glDeleteBuffers(1, &resources.msCopyUniformBlockBuffer);
 
   for(const std::pair<const GLuint, GLuint> &resourceProgram : resources.programs)
   {
     driver->glDeleteProgram(resourceProgram.second);
   }
 
+  for(const auto &pair : resources.copyFramebuffers)
+  {
+    const CopyFramebuffer &cf = pair.second;
+    driver->glDeleteFramebuffers(1, &cf.framebufferId);
+    driver->glDeleteTextures(1, &cf.colorTextureId);
+    driver->glDeleteTextures(1, &cf.dsTextureId);
+    driver->glDeleteTextures(1, &cf.depthTextureId);
+    driver->glDeleteTextures(1, &cf.stencilTextureId);
+    driver->glDeleteTextures(1, &cf.stencilViewId);
+  }
+
   return true;
 }
 
+void CopyMSSample(WrappedOpenGL *driver, const GLPixelHistoryResources &resources,
+                  const CopyFramebuffer &copyFramebuffer, int sampleIdx, int x, int y,
+                  float *pixelDst)
+{
+  GLint savedProgram;
+  driver->glGetIntegerv(eGL_CURRENT_PROGRAM, &savedProgram);
+  GLint savedActiveTexture;
+  driver->glGetIntegerv(eGL_ACTIVE_TEXTURE, &savedActiveTexture);
+  GLint savedShaderStorageBuffer;
+  driver->glGetIntegerv(eGL_SHADER_STORAGE_BUFFER_BINDING, &savedShaderStorageBuffer);
+  GLint savedUniformBuffer;
+  driver->glGetIntegerv(eGL_UNIFORM_BUFFER_BINDING, &savedUniformBuffer);
+
+  driver->glUseProgram(resources.msCopyComputeProgram);
+  GLint srcMSLoc = driver->glGetUniformLocation(resources.msCopyComputeProgram, "srcMS");
+  // GLint srcDepthMSLoc = driver->glGetUniformLocation(resources.msCopyComputeProgram,
+  // "srcDepthMS");
+  // GLint srcStencilMSLoc =
+  //     driver->glGetUniformLocation(resources.msCopyComputeProgram, "srcStencilMS");
```

Comment:
These comments look like old code that can be deleted as it was moved further down.